src/engines/Cesium/index.tsx

```diff
@@ -66,50 +71,61 @@ const Cesium: React.ForwardRefRenderFunction<EngineRef, EngineProps> = (
   ref,
 ) => {
   const {
-    backgroundColor,
     cesium,
-    cameraViewBoundaries,
-    cameraViewOuterBoundaries,
-    cameraViewBoundariesMaterial,
+    engineAPI,
     mouseEventHandles,
     cesiumIonAccessToken,
     context,
-    light,
     layerSelectWithRectEventHandlers,
-    handleMount,
-    handleUnmount,
     handleUpdate,
     handleClick,
-    handleCameraChange,
-    handleCameraMoveEnd,
   } = useHooks({
     ref,
     property,
-    camera,
+    time,
     selectedLayerId,
     selectionReason: layerSelectionReason,
-    isLayerDraggable,
-    isLayerDragging,
     meta,
     layersRef,
     featureFlags,
-    requestingRenderMode,
-    shouldRender,
     timelineManagerRef,
-    cameraForceHorizontalRoll,
     onLayerSelect,
-    onCameraChange,
-    onLayerDrag,
-    onLayerDrop,
     onLayerEdit,
     onLayerSelectWithRectStart,
     onLayerSelectWithRectMove,
     onLayerSelectWithRectEnd,
-    onMount,
     onLayerVisibility,
     onLayerLoad,
   });
 
+  const { sceneLight, sceneBackgroundColor, sceneMsaaSamples, sceneMode } = useViewerProperty({
+    property,
+    cesium,
+  });
+
+  useLayerDragDrop({ cesium, onLayerDrag, onLayerDrop, isLayerDraggable });
+
+  useExplicitRender({ cesium, requestingRenderMode, isLayerDragging, shouldRender, property });
+
+  const {
+    cameraViewBoundaries,
+    cameraViewOuterBoundaries,
+    cameraViewBoundariesMaterial,
+    handleCameraChange,
+    handleCameraMoveEnd,
+    handleMount,
+    handleUnmount,
+  } = useCamera({
+    cesium,
+    property,
+    camera,
+    cameraForceHorizontalRoll,
+    featureFlags,
+    engineAPI,
+    onCameraChange,
+    onMount,
+  });
```

Comment:
Can't we invoke these hooks in `Cesium/hooks.ts`?

Reply:
Yes we can, for sure. I think just style choices, when we dev beta we write it like this sometimes, but it's okey to still wrap all these into Cesium/hooks.